packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,49 @@
+import { PluginOption } from "vite";
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(id: string) {
+  return !!(id.match(/.*@yext\/studio.*/) || id.match(/.*packages\/studio*/));
+}
+
+function isStyleFile(id: string) {
+  return !!id.match(/.*\.s?css.*/);
```

Comment:
We don't support `.scss` files yet, right? Is this just to make sure they're still not added as a tag?

Reply:
Ahh the more I think about it, I'll just change it to `.css` as a skateboard for now.  I originally had some `scss` stuff in the earlier PRs but I removed them, just forgot to change this one! 

Once we get this working nicely with css we can include scss. It hopefully it'll just be a matter of including it in the process later.

Reply:
So in the meantime, if someone imports a `.scss` file, the styling will be applied to Studio?

Reply:
yeah, that would be the case. Do you think it would be preferable if we included scss in this ignore now as well?

Reply:
That would be my preference (since it doesn't seem to difficult to implement). We still won't support applying their styles but at least in the meantime it won't mess with our UI

Reply:
done!